src/Raven.Server/Documents/Document.cs

```diff
@@ -13,6 +14,108 @@ public class TimeSeriesStream
         public IEnumerable<DynamicJsonValue> TimeSeries;
         public string Key;
     }
+
+    public sealed class QueriedDocument : Document, IResetSupport<List<QueriedDocument>>
+    {
+        private static readonly QueriedDocument Empty = new();
+        
+        private int _refCount = 1;
+        
+        
+        private QueriedDocument _parent;
+        private QueriedDocument _child;
+        private List<QueriedDocument> _childs;
```

Comment:
`_referencedDocs` or `_references` maybe? Just to follow our terminology that we use. We usually say that if a document is loaded via a property of another document the it's "a referenced document"